docs/rules/no-classfield-shadowing.md

```diff
@@ -0,0 +1,33 @@
+# Disallows class fields with same name as static properties
+
+Using class fields with same name as static property can have unintended effects.
```

Comment:
Add more detail on the unintended effects? My understanding is that the shadowing of lit properties as class fields causes a runtime error but maybe we can expand this a bit. I think it could also inform the "When Not To Use It" section.

Reply:
The runtime error is:

> The following properties on element simple-greeting will not trigger updates as expected because they are set using class fields: name. Native class fields and some compiled output will overwrite accessors used for detecting changes. See https://lit.dev/msg/class-field-shadowing for more information.

Maybe we just reuse it?